### PR TITLE
fix(stonedb) Valgrind problems(#142)

### PR DESCRIPTION
**Pull request by konghaiya**
## Summary about this PR

Issue Number: close (#137)(#110)(#109)(#108)(#81)(#80)(#67)(#66)(#60)

## Tests Check List

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [X] Bug Fix
- [X] Improvement
- [X] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/stonedb/system/cacheable_item.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ CacheableItem::CacheableItem(char const *owner_name, char const *object_id, int
     // read the configuration parameter
     std::string temp_filename = stonedb_sysvar_cachefolder;
     filename_n_position = temp_filename.length();
-    filename = new char[filename_n_position + 37];  // "...path.../XXXXXXnnnnnnAAAAAAAABBBBBBBB.sdb_tmp"
+    filename = new char[filename_n_position + 38];  // "...path.../XXXXXXnnnnnnAAAAAAAABBBBBBBB.sdb_tmp"
```
Comment (RingsC) [inline storage/stonedb/system/cacheable_item.cpp]: do initialization after allocation

File: storage/stonedb/system/cacheable_item.cpp [hunk 2]
```diff
@@ -45,7 +45,7 @@ CacheableItem::CacheableItem(char const *owner_name, char const *object_id, int
 
   if (filename == NULL) {
     // if the temporary path is not set, use the current folder
-    filename = new char[36];  // "XXXXXXnnnnnnAAAAAAAABBBBBBBB.sdb_tmp"
+    filename = new char[37];  // "XXXXXXnnnnnnAAAAAAAABBBBBBBB.sdb_tmp"
```
Comment (RingsC) [inline storage/stonedb/system/cacheable_item.cpp]: same as above

File: storage/stonedb/system/cacheable_item.cpp (hunk 1 above)
Comment (RingsC) [inline storage/stonedb/system/cacheable_item.cpp]: Do not use magic number

File: storage/stonedb/system/cacheable_item.cpp (hunk 2 above)
Comment (RingsC) [inline storage/stonedb/system/cacheable_item.cpp]: Do not use magic number

File: storage/stonedb/system/cacheable_item.cpp [hunk 3]
```diff
@@ -34,7 +34,7 @@ CacheableItem::CacheableItem(char const *owner_name, char const *object_id, int
     // read the configuration parameter
     std::string temp_filename = stonedb_sysvar_cachefolder;
     filename_n_position = temp_filename.length();
-    filename = new char[filename_n_position + 37];  // "...path.../XXXXXXnnnnnnAAAAAAAABBBBBBBB.sdb_tmp"
+    filename = new char[filename_n_position + 38];  // "...path.../XXXXXXnnnnnnAAAAAAAABBBBBBBB.sdb_tmp"
     std::strcpy(filename, temp_filename.c_str());
```
Comment (RingsC) [inline storage/stonedb/system/cacheable_item.cpp]: use strncpy, not this. to make it string cpy safty.

**Review by RingsC — CHANGES_REQUESTED** [review]

**Review by hustjieke — APPROVED** [review]

**Review by RingsC — APPROVED** [review]
LGTM

**Pull request merged**